File: mobile/android/app/src/main/java/com/infy/messenger/feature/auth/domain/AuthRepository.kt
package com.infy.messenger.feature.auth.domain

/**
 * Доменный контракт авторизации. UI/ViewModel зависят только от него.
 * Реализация ([com.infy.messenger.feature.auth.data.AuthRepositoryImpl]) бросает
 * [com.infy.messenger.core.network.ApiError] при сбоях.
 */
interface AuthRepository {
    suspend fun register(
        username: String,
        nickname: String,
        password: String,
        email: String?,
    ): AuthSession

    suspend fun login(username: String, password: String): AuthSession

    suspend fun requestPasswordReset(email: String)

    suspend fun logout()
}

File: mobile/ios/Infy/Features/Authentication/Domain/AuthRepository.swift
import Foundation

protocol AuthRepository: Sendable {
    func login(username: String, password: String) async throws -> AuthSession
    func register(username: String, nickname: String, password: String, email: String?, birthdate: String?) async throws -> AuthSession
    func forgotPassword(email: String) async throws
    func validateResetToken(_ token: String) async throws -> ResetPasswordValidation
    func resetPassword(token: String, password: String) async throws
    func logout() async throws
}

